apps/web/app/unsubscribe/[userId]/unsubscribe-form.tsx

```diff
@@ -0,0 +1,30 @@
+"use client";
+
+import { Button } from "@workspace/ui/components/button";
+import Link from "next/link";
+import { useRouter } from "next/navigation";
+
+export function UnsubscribeForm({ userId }: { userId: string }) {
+  const router = useRouter();
+
+  const handleUnsubscribe = () => {
+    router.push(`/unsubscribe/${userId}?confirmed=true`);
+  };
```

Comment:
non use Server Action to do this, look at other server action and use it with useMutation,look at code example !!!